lib/sles4sap_publiccloud.pm

```diff
@@ -741,4 +794,145 @@ sub create_hana_vars_section {
     return (\%hana_vars);
 }
 
+=head2 setup_azure_native_fencing_permissions
+
+    setup_azure_native_fencing_permissions(vmname=>$vm_name,
+        subscription_id=>$subscription_id,
+        resource_group=>$resource_group,
+        role=>$role);
+
+    Sets up managed identity (MSI) by enabling system assigned identity and
+        role - role to be assigned
+        resource_group - resource group resource belongs to
+        subscription_id - valid azure subscription
+=cut
+
+sub setup_azure_native_fencing_permissions {
+    my (%args) = @_;
+    my $vm_name = $args{vm_name};
+    my $subscription_id = $args{subscription_id};
+    my $resource_group = $args{resource_group};
+    my $role = 'Virtual Machine Contributor';
+
+    foreach ('vm_name', 'subscription_id', 'resource_group') {
+        croak "Missing argument: '$_'" unless defined($args{$_});
+    }
+
+    my $vm_id = azure_enable_system_assigned_identity($vm_name, $resource_group);
+    azure_assign_role(assignee => $vm_id, role => $role, subscription_id => $subscription_id, resource_group => $resource_group);
+    return 1;
+}
+
+=head2 azure_enable_system_assigned_identity
+
+    azure_enable_system_assigned_identity($vm_name, $resource_group);
+
+    Enables 'System assigned identity' for specified VM.
+    Returns 'systemAssignedIdentity' ID.
+
+=cut
+
+sub azure_enable_system_assigned_identity {
+    my ($vm_name, $resource_group) = @_;
+    croak "Missing 'vm_name' or 'resource_group argument'" unless ($vm_name and $resource_group);
+
+    my $az_cmd = "az vm identity assign";
+    my $az_args = "--only-show-errors -g '$resource_group' -n '$vm_name' --query 'systemAssignedIdentity' -o tsv";
+    my $identity_id = script_output(join(' ', $az_cmd, $az_args));
+    croak 'Returned output does not match ID pattern' unless
+      $identity_id =~ /[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}/;
+    return $identity_id;
+}
+
+=head2 azure_assign_role
+
+    azure_assign_role( assignee=>$assignee, role=>$role, subscription_id=>$subscription_id, resource_group=>$resource_group )
+
+    Assigns defined role to 'assignee' (user, vm, etc...) using subscription id.
+     assignee - UUID for the resource (VM in this case)
+     role - role to be assigned
+     resource_group - resource group resource belongs to
+     subscription_id - valid azure subscription
+
+=cut
+
+sub azure_assign_role {
```

Comment:
This one seems to me almost pure az composition and execution. If it is so I suggest to:

    rename it to qesap_az_assign_role
    move it to lib/qesapdeployment.pm
    test it in t/15_qesap_azure.t

Reply:
Moved, thanks.